tensorflow_addons/activations/gelu.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implementing GELU activation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@tf.function
+@keras_utils.register_keras_custom_object
+def gelu(x):
+    """Gaussian Error Linear Unit.
+
+    A smoother version of ReLU generally used 
+    in the BERT or BERT architecture based models.
+    Original paper: https://arxiv.org/abs/1606.08415
+    
+    Args:
+        x: float Tensor to perform activation.
+    Returns:
+        `x` with the GELU activation applied.
+    """
+
+    x = tf.convert_to_tensor(x, dtype=tf.float32)
```

Comment:
Do we really need to cast to float32? I mean maybe float16 and float64 are friendly to the following computation.

Reply:
Most of the operations in TF happens in `float32`. `float64` is rarely used and for `fp16`, I need to check all the changes I have to make. If I am right, then all other floating operations have to be in `fp16` then.